src/cs_transmog.cpp

```diff
@@ -56,6 +68,163 @@ class transmog_commandscript : public CommandScript
         player->UpdateObjectVisibility();
         return true;
     }
+
+    static bool HandleAddTransmogItem(ChatHandler* handler, Optional<PlayerIdentifier> player, ItemTemplate const* itemTemplate)
+    {
+        if (!sTransmogrification->GetUseCollectionSystem())
+            return true;
+
+        if (!sObjectMgr->GetItemTemplate(itemTemplate->ItemId))
+        {
+            handler->PSendSysMessage(LANG_COMMAND_ITEMIDINVALID, itemTemplate->ItemId);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (!player)
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player || !player->IsConnected())
+            return false;
```

Comment:
```suggestion
        if (!player) {
            player = PlayerIdentifier::FromTargetOrSelf(handler);
            return false;
        }
        if (!player->IsConnected()) {
            return false;
        }
```

Reply:
Player gets redefined after the first `!player`, that's why player is getting checked again as well and not returned right away.

Reply:
oh, ok nice 👍